src/audio/tdfb/tdfb.c

```diff
@@ -117,14 +162,39 @@ static void tdfb_free_delaylines(struct tdfb_comp_data *cd)
 		fir[i].delay = NULL;
 }
 
+static void *tdfb_filter_seek(struct sof_tdfb_config *config, int num_filters)
+{
+	struct sof_fir_coef_data *coef_data;
+	int i;
+	uint8_t *coefp = (uint8_t *)ASSUME_ALIGNED(&config->data[0], 2); /* 2 for 16 bits data */
+
+	/* Note: FIR coefficients are int16_t. An uint_8 type pointer coefp
+	 * is used for jumping in the flexible array member structs coef_data.
+	 */
+	for (i = 0; i < num_filters; i++) {
+		coef_data = (struct sof_fir_coef_data *)coefp;
+		coefp += sizeof(*coef_data) + sizeof(int16_t) * coef_data->length;
+	}
+
+	return (void *)coefp;
```

Comment:
in C you don't need to type-cast between `void *` and other pointers (unless there's `const` or something else, which shouldn't be done anyway), but if you change everything to `int16_t *` all the types will match anyway